R/fit.R

```diff
@@ -247,3 +263,19 @@ fit_model_approximate <- function(args, future = FALSE, id = "stan") {
   }
   return(fit)
 }
+
+#' Calculate number of post-warmup iterations and samples based on the backend
+#' @keywords internal
+#' @inheritParams fit_model_with_nuts
+create_logging_sampler_values <- function(args) {
```

Comment:
This looks good. Would it be better to make it less sideeffectie though and instead pass in backend as the arg and have it output jhust he list of things you need for reporting as its own thing i.e not overwrite args?

Reply:
Sure. Sounds good.

Reply:
How about now? 6379b3d322528fd67e50ff86c294a396e328d6ad